src/test/java/com/amazon/opendistroforelasticsearch/security/InitializationIntegrationTests.java

```diff
@@ -198,6 +198,29 @@ public void testDefaultConfig() throws Exception {
         Assert.assertEquals(HttpStatus.SC_OK, rh.executeGetRequest("", encodeBasicHeader("admin", "admin")).getStatusCode());
     }
 
+    @Test
+    public void testInvalidDefaultConfig() throws Exception {
+        String defaultInitDirectory = System.getProperty("security.default_init.dir");
+        try {
+            System.setProperty("security.default_init.dir", new File("./src/test/resources/invalid_config").getAbsolutePath());
+            final Settings settings = Settings.builder()
+                    .put(ConfigConstants.OPENDISTRO_SECURITY_ALLOW_DEFAULT_INIT_SECURITYINDEX, true)
+                    .build();
+            setup(Settings.EMPTY, null, settings, false);
+            RestHelper rh = nonSslRestHelper();
+            Thread.sleep(10000);
+            Assert.assertEquals(HttpStatus.SC_SERVICE_UNAVAILABLE, rh.executeGetRequest("", encodeBasicHeader("admin", "admin")).getStatusCode());
+
+            System.setProperty("security.default_init.dir", new File(defaultInitDirectory).getAbsolutePath());
+            restart(Settings.EMPTY, null, settings, false);
+            rh = nonSslRestHelper();
+            Thread.sleep(10000);
+            Assert.assertEquals(HttpStatus.SC_OK, rh.executeGetRequest("", encodeBasicHeader("admin", "admin")).getStatusCode());
+        } finally {
+            System.setProperty("security.default_init.dir", new File(defaultInitDirectory).getAbsolutePath());
```

Comment:
Check `security.default_init.dir` for `null` and set it only if it is not null. It can be set to an invalid path (for whatever reason) causing `IOException`.

Reply:
If `defaultInitDirectory` is null and in which case we don't set the concerned property, then all remaining tests will fail.
I was thinking I could just copy over from ClusterHelper.java which is initially setting the property? Or extract it to a function and use it in both the places.

Reply:
The goal here is to preserve `"security.default_init.dir"` property value independent from what `ClusterHelper` does. If it was not set (`null`), it needs to be unset, otherwise it needs to be set to the original value.

Reply:
Got it, makes sense. Will set it to original only if it is non-null.